build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -18,14 +18,16 @@ import 'package:build_runner/src/watcher/collect_changes.dart';
 import 'package:build_runner/src/watcher/delete_writer.dart';
 import 'package:build_runner/src/watcher/graph_watcher.dart';
 import 'package:build_runner/src/watcher/node_watcher.dart';
-import 'package:build_runner_core/build_runner_core.dart';
+import 'package:build_runner_core/build_runner_core.dart'
+    hide BuildResult, BuildStatus;
+import 'package:build_runner_core/build_runner_core.dart' as core;
```

Comment:
how about `as core show BuildResult, BuildStatus;`

That way we don't accidentally have the same type with and without a prefix anywhere.